### PR TITLE
[HK] updates validatedBounds Tests

### PR DESCRIPTION
**Pull request by fgdrf**
* uses SubMonitor rather than deprecated SubProgressMonitor
* new Tests for MetricsFactory
* removes sysout

Signed-off-by: Frank Gasdorf <fgdrf@users.sourceforge.net>

### CODE REVIEW

**Comment by fgdrf** [comment]
@nprigour Are you kind to review this changeset as well? Thanks in advance!

File: plugins/org.locationtech.udig.render.feature.basic.tests/src/org/locationtech/udig/render/internal/feature/basic/BasicFeatureMetricsFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        public <T> boolean canResolve(Class<T> adaptee) {
+            if (resolveClass!=null && adaptee != null && adaptee.getName().equals(resolveClass.getName())) {
+                return true;
+            }
+            return false;
+        }
+    }
+
+    @Test
+    public void giveImageMoasicPriorityOverShapefile() throws Exception {
```
Comment (nprigour) [inline plugins/org.locationtech.udig.render.feature.basic.tests/src/org/locationtech/udig/render/internal/feature/basic/BasicFeatureMetricsFactoryTest.java]: correct the typo in the method name

File: plugins/org.locationtech.udig.render.feature.basic.tests/src/org/locationtech/udig/render/internal/feature/basic/BasicFeatureMetricsFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        IRenderContext mockedRenderContect = EasyMock.createNiceMock(IRenderContext.class);
```
Comment (nprigour) [inline plugins/org.locationtech.udig.render.feature.basic.tests/src/org/locationtech/udig/render/internal/feature/basic/BasicFeatureMetricsFactoryTest.java]: correct the typo in the variable name

File: plugins/org.locationtech.udig.render.feature.basic.tests/src/org/locationtech/udig/render/internal/feature/basic/BasicFeatureMetricsFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        
+        assertFalse(new BasicFeatureMetricsFactory().canRender(mockedRenderContect));
+        
+        EasyMock.verify(mockedGeoResource, mockedRenderContect);
+        
+    }
+
+    @Test
+    public void acceptOnFeatureSource() throws Exception {
+        IRenderContext mockedRenderContect = EasyMock.createNiceMock(IRenderContext.class);
```
Comment (nprigour) [inline plugins/org.locationtech.udig.render.feature.basic.tests/src/org/locationtech/udig/render/internal/feature/basic/BasicFeatureMetricsFactoryTest.java]: correct the typo in the variable name

**Comment by nprigour** [comment]
I have no comments regarding the implementation. Just some minor comments regarding verbalization and naming :)

File: plugins/org.locationtech.udig.render.feature.basic.tests/src/org/locationtech/udig/render/internal/feature/basic/BasicFeatureMetricsFactoryTest.java (hunk 1 above)
Comment (fgdrf) [reply plugins/org.locationtech.udig.render.feature.basic.tests/src/org/locationtech/udig/render/internal/feature/basic/BasicFeatureMetricsFactoryTest.java]: Implementation doesn't provide a Metrics, if the underlaying resource is an AbstractGridCoverage2DReader

**Pull request merged**